src/ralph/sim_cards/models.py

```diff
@@ -0,0 +1,130 @@
+from dj.choices import Choices
+from django.conf import settings
+from django.core.validators import (
+    MaxLengthValidator,
+    MinLengthValidator,
+    RegexValidator
+)
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.back_office.models import Warehouse
+from ralph.lib.mixins.models import (
+    AdminAbsoluteUrlMixin,
+    NamedMixin,
+    TimeStampMixin
+)
+from ralph.lib.transitions.fields import TransitionField
+from ralph.lib.transitions.models import TransitionWorkflowBase
+
+
+PUK_CODE_VALIDATORS = [
+    MinLengthValidator(5),
+    RegexValidator(
+        regex='^\d+$',
+        message=_('Required numeric characters only.')
+    ),
+]
+
+
+PIN_CODE_VALIDATORS = [
+    MinLengthValidator(4),
+    RegexValidator(
+        regex='^\d+$',
+        message=_('Required numeric characters only.')
+    ),
+]
+
+
+class SIMCardStatus(Choices):
+    _ = Choices.Choice
+
+    new = _("new")
+    in_progress = _("in progress")
+    waiting_for_release = _("waiting for release")
+    used = _("in use")
+    damaged = _("damaged")
+    liquidated = _("liquidated")
+    free = _("free")
+    reserved = _("reserved")
+    loan_in_progress = _("loan in progress")
+    return_in_progress = _("return in progress")
+    in_quarantine = _("in quarantine")
+
+
+class CellularCarrier(AdminAbsoluteUrlMixin, NamedMixin, models.Model):
+    pass
+
+
+class SIMCard(
+    AdminAbsoluteUrlMixin,
+    TimeStampMixin,
+    models.Model,
+    metaclass=TransitionWorkflowBase
+):
+    pin1 = models.CharField(
+        max_length=8, null=True, blank=True,
+        help_text=_('Required numeric characters only.'),
+        validators=PIN_CODE_VALIDATORS
+    )
+    puk1 = models.CharField(
+        max_length=16, help_text=_('Required numeric characters only.'),
+        validators=PUK_CODE_VALIDATORS
+    )
+    pin2 = models.CharField(
+        max_length=8, null=True, blank=True,
+        help_text=_('Required numeric characters only.'),
+        validators=PIN_CODE_VALIDATORS
+    )
+    puk2 = models.CharField(
+        max_length=16, null=True, blank=True,
+        help_text=_('Required numeric characters only.'),
+        validators=PUK_CODE_VALIDATORS)
+    carrier = models.ForeignKey(
+        CellularCarrier, on_delete=models.PROTECT,
+    )
+    card_number = models.CharField(
+        max_length=22, unique=True,
+        validators=[
+            MinLengthValidator(1),
+            MaxLengthValidator(22),
+            RegexValidator(
+                regex='^\d+$',
+                message=_('Required numeric characters only.'),
+            )
+        ]
+    )
+    phone_number = models.CharField(
+        max_length=16, unique=True, help_text=_('ex. +48668234346'),
```

Comment:
This looks like a valid phone number. Maybe it would be better to use the number from regex validator below or some other fictitious number.

Reply:
done